File: seed/pydantic-v2/trace/resources/commons/types/binary_tree_value.py
from typing import Dict, Optional

from core.datetime_utils import serialize_datetime
from dt import datetime
from resources.commons.types.binary_tree_node_value import BinaryTreeNodeValue

from pydantic import BaseModel


class BinaryTreeValue(BaseModel):
    root: Optional[str] = None
    nodes: Dict[str, BinaryTreeNodeValue]

    class Config:
        frozen = True
        smart_union = True
        json_encoders = {datetime: serialize_datetime}

File: seed/ruby-sdk-v2/trace/lib/seed/commons/binary_tree_value.rb
# frozen_string_literal: true

module Seed
    module Types
        class BinaryTreeValue < Internal::Types::Model
            field :root, String, optional: true, nullable: false
            field :nodes, Internal::Types::Hash[String, Seed::Commons::BinaryTreeNodeValue], optional: false, nullable: false

    end
end
